### PR TITLE
[ADAPTER] customer.io 

### PR DESCRIPTION
**Pull request by lucacorti**
Hi,

I need to send emails via customer.io transactional email API. This implementation might have some rough edges but is working for me.

### CODE REVIEW

File: lib/swoosh/adapters/customer_io.ex [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    Map.put(body, :headers, headers)
+  end
+
+  defp prepare_provider_options_body_fields(body, %Email{provider_options: provider_options}) do
+    Map.merge(body, Map.take(provider_options, @provider_options_body_fields))
+  end
+
+  defp email_item({"", address}), do: address
+  defp email_item({name, address}), do: name <> " <" <> address <> ">"
+  defp email_item(address), do: address
```
Comment (princemaple) [inline lib/swoosh/adapters/customer_io.ex:213]: https://github.com/swoosh/swoosh/blob/main/lib/swoosh/email/render.ex

Does this work?

**Comment by princemaple** [comment]
Fairly good looking PR, thanks! Just that one question for now.

File: lib/swoosh/adapters/customer_io.ex (hunk 1 above)
Comment (lucacorti) [reply lib/swoosh/adapters/customer_io.ex]: Yes, I just copied the code over from the Sendgrid adapter, fixed.

**Comment by princemaple** [comment]
🎉Thank you!

**Pull request merged**